content/posts/2023-04-05-microshift-on-crc.md

```diff
@@ -0,0 +1,34 @@
+---
+title: " Introducing MicroShift preset for CRC"
+date: 2023-04-05T07:12:10+05:30
+---
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
+Or do you need to deploy to a small form factor or edge computing environment? 
+If so, you'll be happy to hear about [MicroShift](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/index),
+a lightweight version of OpenShift Kubernetes that's optimized for resource-limited environments.
+
+Currently, OpenShift preset provided by CRC is required 9GB of RAM and 4CPUs.
+With the release of CRC-2.16, users can now use the MicroShift preset, which only requires 4GB of RAM and 2 CPUs.
+
+What makes MicroShift different from OpenShift Kubernetes? MicroShift is designed for small form factor
+and edge computing environments, which means it's optimized for lower resource requirements. MicroShift
+uses a subset of the OpenShift Kubernetes components and removes some of the features that aren't needed in these environments.
+
+[Key differences from OpenShift Kubernetes](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture)
+
+To get started with MicroShift, you'll need to download and install the [latest version]() of CRC.
+If you already have a CRC instance, delete it using the `crc delete` command. 
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+
+After you start CRC, you'll receive the `kubeconfig` file location, which you can export to connect with the cluster. 
+
+To try out MicroShift, we've created a sample demo repository that you can find on [GitHub](https://github.com/praveenkumar/simple-go-server).
+It includes a workload that you can use to test out the MicroShift preset.
```

Comment:
I was thinking we could also provide an image that can be directly deployed on the cluster to make sure it works, and then to provide instructions how to build that image for those that are interested in this.

Reply:
that something we can do that on demo repo if we need to update it. One think I have in mind to add step how microshift preset can be added as github action.